txtorcon/circuit.py

```diff
@@ -275,3 +275,25 @@ def __str__(self):
         path = ' '.join([x.ip for x in self.path])
         return "<Circuit %d %s [%s] for %s>" % (self.id, self.state, path,
                                                 self.purpose)
+
+class CircuitBuildTimedOutError(Exception):
+    """
+    This exception is thrown when using `timed_circuit_build`
+    and the circuit build times-out.
+    """
+
+def build_timeout_circuit(tor_state, clock, path, timeout):
+    """
+    returns a deferred which fires when the
+    circuit build succeeds or fails to build.
+    CircuitBuildTimedOutError will be raised unless we
+    receive a circuit build result within the `timeout` duration.
+    """
+    d = tor_state.build_circuit(path, using_guards=False)
+    clock.callLater(timeout, d.cancel)
```

Comment:
...also, it's probably best to "pass along" this kwarg:

``build_circuit(tor_state, reactor, path, using_guards=False, timeout=None)``